machine/dell/dell_s4000_c2338/i2ctools/i2ctools-sys-eeprom-platform.patch

```diff
@@ -0,0 +1,255 @@
++    i2c_bus = get_i2c_bus();
++    memset(i2c_bus_path, 0, MAX_EEPROM_I2C_BUS_PATH);
++    snprintf(i2c_bus_path, MAX_EEPROM_I2C_BUS_PATH, "/dev/i2c-%d", i2c_bus);
++
+     for (i = 0; i < len; i++) {
+-	if (eeprom_open(SYS_EEPROM_I2C_DEVICE, SYS_EEPROM_I2C_ADDR,
++	if (eeprom_open(i2c_bus_path, i2c_addr,
+ 			EEPROM_TYPE_8BIT_ADDR, &e)) {
+ 	    printf("ERROR: Cannot open I2C device\n");
+ 	    return -1;
+ 	}
+ 	eeprom_write_byte(&e, addr, *c);
++	usleep(5000);
+ 	eeprom_close(&e);
+ 	c++; addr++;
+     }
+diff --git a/sys_eeprom/onie_tlvinfo.h b/sys_eeprom/onie_tlvinfo.h
+index 5714f37..da8ab3a 100644
+--- a/sys_eeprom/onie_tlvinfo.h
++++ b/sys_eeprom/onie_tlvinfo.h
+@@ -170,3 +170,5 @@ bool tlvinfo_delete_tlv(u_int8_t * eeprom, u_int8_t code);
+ bool tlvinfo_add_tlv(u_int8_t * eeprom, int tcode, char * strval);
+ bool tlvinfo_decode_tlv(u_int8_t *eeprom, u_int8_t tcode, char* value);
+ void show_tlv_code_list(void);
++int get_i2c_addr(void);
++int get_i2c_bus (void);
+diff --git a/sys_eeprom/sys_eeprom.c b/sys_eeprom/sys_eeprom.c
+index a77837f..2c57a58 100644
+--- a/sys_eeprom/sys_eeprom.c
++++ b/sys_eeprom/sys_eeprom.c
+@@ -5,6 +5,9 @@
+ #include "onie_tlvinfo.h"
+ 
+ static u_int8_t eeprom[SYS_EEPROM_SIZE];
++static int i2c_addr = SYS_EEPROM_I2C_ADDR;
++static int i2c_bus = 0;
++
+ /*
+  *  This macro defines the sys_eeprom command line command.
+  */
+@@ -12,10 +15,14 @@ cmd_usage()
+ {
+     static const char *usage =
+ 	"Display and program the system EEPROM data block.\n"
+-	"Usage: sys-eeprom [-h][-l] [-e] [-s <code>=<value>,...]\n"
++	"Usage: sys-eeprom [-h][-l] [-e] [-b bus] [-a address] [-s <code>=<value>,...]\n"
+ 	"   With no arguments display the EEPROM contents.\n"
+ 	"   -h --help\n"
+ 	"      Display usage\n"
++	"   -a --addr <address>\n"
++	"      Set the I2C bus address. \n"
++	"   -b --bus <bus>\n"
++	"      Set the I2C device bus. \n"
+ 	"   -l --list\n"
+ 	"      List the understood TLV codes and names.\n"
+ 	"   -e --erase\n"
+@@ -38,24 +45,56 @@ int main(int argc, char * const argv[])
+     int count = 0;
+     int err = 0;
+     int update = 0;
++    int skip_usage = 0;
+     char *value, *subopts, *tname;
+     int index, c, i, option_index, tcode;
+     char tlv_value[TLV_DECODE_VALUE_MAX_LEN];
++    char i2c_sw_sel_cmd[SYS_EEPROM_I2C_SW_CMD_SZ];
++    int idx;
++    bool address_present = 0;
+ 
+     const size_t tlv_code_count = sizeof(tlv_code_list) /
+ 	sizeof(tlv_code_list[0]);
+ 
+     char *tokens[tlv_code_count + 1];
+-    const char *short_options = "hels:g:";
++    const char *short_options = "hels:g:a:b:";
+     const struct option long_options[] = {
+ 	{"help",    no_argument,          0,    'h'},
+ 	{"list",    no_argument,          0,    'l'},
+ 	{"erase",   no_argument,          0,    'e'},
+ 	{"set",     required_argument,    0,    's'},
+ 	{"get",     required_argument,    0,    'g'},
++	{"addr",    required_argument,    0,    'a'},
++	{"bus",     required_argument,    0,    'b'},
```

Comment:
Nice.  I think the -a and -b is a cool feature we should add to the *core* of the EEPROM code, as it could be useful for others as well.  It would be straight forward to split off this feature into it a stand alone "feature patch" and have it not be DELL S4000 specific.

Now I have a bigger problem, one that predates your participation in the project.  Please see these threads from the mailing list:

http://lists.opencompute.org/pipermail/opencompute-onie/2014-September/000103.html
http://lists.opencompute.org/pipermail/opencompute-onie/2014-September/000150.html

TL;DR -- please do not use the i2ctools version of the EEPROM code, it is deprecated.  Use the busybox version.

I will write more on the mailing list about this topic to reinforce the message for everyone.